halo2-ecc/src/ecc/mod.rs

```diff
@@ -1014,6 +1036,36 @@ impl<'chip, F: BigPrimeField, FC: FieldChip<F>> EccChip<'chip, F, FC>
 where
     FC: Selectable<F, FC::FieldPoint>,
 {
+    pub fn sum_safe<C>(
```

Comment:
I think we can call your version the normal `sum` and the old `sum` rename to `unsafe_sum`; first please go through all current use cases of `sum` and see if they can use `unsafe_sum` or if they should actually use `sum` (safe) instead